main.py

```diff
@@ -553,11 +553,59 @@ def print_all_stats(commits_stats: dict, lines_stats: dict, contributors_stats:
 
 
 def main():
-    use_cloc = input("Do you want to use cloc (C) or wc (W) to count SLOC? c/W ").lower() == "c"
-    get_commits = input("Do you want to get the commits stats? It may take a long time due to GitHub servers updating "
-                        "their cache. y/N ").lower() == "y"
-    get_lines = input("Do you want to get the SLOC stats? It may take a long time since it has to clone each repository. y/N ").lower() == "y"
-    generate_graphs = input("Do you want to generate graphs for the statistics? y/N ").lower() == 'y'
+    import argparse
+
+    parser = argparse.ArgumentParser(description="S.A.R.D.I.N.A. - Statistiche Amabili Rendimento Degli Informatici Nell'Anno")
+
+    cloc_group = parser.add_argument_group('Software to use to count lines of code').add_mutually_exclusive_group(required=False)
+    cloc_group.add_argument('--cloc', action='store_true', default=None, help="Use CLOC to count SLOC.")
+    cloc_group.add_argument('--wc', action='store_true', default=None, help="Use WC to count SLOC.")
+
+    commits_group = parser.add_argument_group('Count contributions to all repositories').add_mutually_exclusive_group(required=False)
+    commits_group.add_argument('--commits', action='store_true', default=None, help="Count commits.")
+    commits_group.add_argument('--no-commits', action='store_true', default=None, help="Do not count commits.")
+
+    sloc_group = parser.add_argument_group('Count SLOC (source lines of code) of repositories').add_mutually_exclusive_group(required=False)
+    sloc_group.add_argument('--sloc', action='store_true', default=None, help="Count SLOC.")
+    sloc_group.add_argument('--no-sloc', action='store_true', default=None, help="Do not count SLOC.")
+
+    graph_group = parser.add_argument_group('Generate graphs for the gathered statistics').add_mutually_exclusive_group(required=False)
+    graph_group.add_argument('--graphs', action='store_true', default=None, help="Generate graphs.")
+    graph_group.add_argument('--no-graphs', action='store_true', default=None, help="Do not generate graphs.")
+
+    parser.add_argument('-p', '--ping', required=False, default=None, action='store_true',
+                                        help='Only ping GitHub servers and exit with 0 if they can be reached.')
+
+    args = parser.parse_args()
+
+    if args.ping:
+        try:
+            response = requests.get('https://github.com')
+            exit(0)
+        except requests.exceptions.ConnectionError:
```

Comment:
Sorry but this is not what I intended with "daemon" or "cron" mode. The ping option should not only ping GitHub servers, but ping the APIs for lines and commits for all repos, as we have always done, but without saving anything. It should be a shortcut to `main.py --commits --sloc --no-graphs` (I didn't specify `--cloc` or `--wc` because in this case their use is pointless, since we won't run either.

Reply:
Should we also try to clone a repo or have faith that if `github.com` responds to requests also cloning is going to work? If yes, should we attempt to clone _all_ repos or just one (or a few)?

Reply:
We don't need to clone any repositories. The "ping" to the APIs is to trigger GitHub's servers to re-generate stats (if needed, they reply with 202 in this case)

Reply:
So it would be fine to replace `--sloc` with `--no-sloc` in the original review? As all APIs should be reached and polled regardless

Reply:
Should be fixed with 9e995c2